### PR TITLE
Display detailed error in FsCacheHandler constructor when cachefiles device not ready

### PR DESCRIPTION
**Pull request by benjaminhuanghuang**
Display detailed error when construct FsCacheHandler with a nonexistent path

Files: fs_cache.rs

Signed-off-by: Benjamin Huang <benjaminhuanghuang@gmail.com>

### CODE REVIEW

**Comment by yqleng1987** [comment]
@benjaminhuanghuang , a new test job has been submitted. Please wait in patience.

**Comment by yqleng1987** [comment]
@benjaminhuanghuang , The CI test is completed, please check result:
<table>
    <tr>
        <th colspan="2">Test Case</th><th>Test Result</th>
    </tr>
    <tr>
        <td colspan="2">merge-target-branch</td><td><a href=https://tone.openanolis.cn/ws/nrh4nnio/test_result/9786>:white_check_mark:SUCCESS</a></td>
    </tr>
    <tr>
        <td colspan="2">build-docker-image</td><td><a href=https://tone.openanolis.cn/ws/nrh4nnio/test_result/9786>:white_check_mark:SUCCESS</a></td>
    </tr>
    <tr>
        <td colspan="2">compile-nydus</td><td><a href=https://tone.openanolis.cn/ws/nrh4nnio/test_result/9786>:x:FAIL</a></td>
    </tr>
</table>

Sorry, your test job failed. Please get the details in the link.

File: src/bin/nydusd/fs_cache.rs [hunk 1]
```diff
@@ -250,7 +250,12 @@ impl FsCacheHandler {
             .write(true)
             .read(true)
             .create(false)
-            .open(path)?;
+            .open(path)
+            .map_err(|e| {
+              error!("Failed to open path {}. {}", path, e);
```
Comment (imeoer) [inline src/bin/nydusd/fs_cache.rs:255]: `Failed to open cachefiles device {}: {}, forget "modprobe cachefiles" ?`

**Comment by yqleng1987** [comment]
@benjaminhuanghuang , your pull request has been updated. A new test job will be submitted. Please wait in patience.

**Comment by yqleng1987** [comment]
@benjaminhuanghuang , the test job has been submitted. Please wait in patience.

**Comment by yqleng1987** [comment]
@benjaminhuanghuang , your pull request has been updated, but a test job is already running. A new test job wil be submitted after the current job is finished.

**Comment by yqleng1987** [comment]
@benjaminhuanghuang , your test job is running now. Please wait in patience.

**Comment by yqleng1987** [comment]
@benjaminhuanghuang , The CI test is completed, please check result:
<table>
    <tr>
        <th colspan="2">Test Case</th><th>Test Result</th>
    </tr>
    <tr>
        <td colspan="2">merge-target-branch</td><td><a href=https://tone.openanolis.cn/ws/nrh4nnio/test_result/9844>:white_check_mark:SUCCESS</a></td>
    </tr>
    <tr>
        <td colspan="2">build-docker-image</td><td><a href=https://tone.openanolis.cn/ws/nrh4nnio/test_result/9844>:white_check_mark:SUCCESS</a></td>
    </tr>
    <tr>
        <td colspan="2">compile-nydus</td><td><a href=https://tone.openanolis.cn/ws/nrh4nnio/test_result/9844>:white_check_mark:SUCCESS</a></td>
    </tr>
    <tr>
        <td colspan="2">compile-ctr-remote</td><td><a href=https://tone.openanolis.cn/ws/nrh4nnio/test_result/9844>:white_check_mark:SUCCESS</a></td>
    </tr>
    <tr>
        <td colspan="2">compile-nydus-snapshotter</td><td><a href=https://tone.openanolis.cn/ws/nrh4nnio/test_result/9844>:white_check_mark:SUCCESS</a></td>
    </tr>
    <tr>
        <td colspan="2">start-nydus-snapshotter-config-containerd</td><td><a href=https://tone.openanolis.cn/ws/nrh4nnio/test_result/9844>:white_check_mark:SUCCESS</a></td>
    </tr>
    <tr>
        <td colspan="2">run-container-with-nydus-image</td><td><a href=https://tone.openanolis.cn/ws/nrh4nnio/test_result/9844>:white_check_mark:SUCCESS</a></td>
    </tr>
</table>

Sorry, the current test job result is not valid, because your pull request has been updated, and a new test job will be submitted.

File: src/bin/nydusd/fs_cache.rs [hunk 2]
```diff
@@ -250,7 +250,15 @@ impl FsCacheHandler {
             .write(true)
             .read(true)
             .create(false)
-            .open(path)?;
+            .open(path)
+            .map_err(|e| {
+                error!(
+                    "Failed to open cachefiles device {}. {}. Please try `modprobe cachefiles`",
```
Comment (imeoer) [inline src/bin/nydusd/fs_cache.rs:256]: "Failed to open cachefiles device {}:" is better. :)

**Review by imeoer — APPROVED** [review]

File: src/bin/nydusd/fs_cache.rs (hunk 2 above)
Comment (hsiangkao) [reply src/bin/nydusd/fs_cache.rs:256]: Yeah, I think `modprobe cachefiles` is no needed to be seen here, there might be other reasons.

**Comment by yqleng1987** [comment]
@hsiangkao , the test job has been submitted. Please wait in patience.

**Comment by yqleng1987** [comment]
@hsiangkao , The CI test is completed, please check result:
<table>
    <tr>
        <th colspan="2">Test Case</th><th>Test Result</th>
    </tr>
    <tr>
        <td colspan="2">merge-target-branch</td><td><a href=https://tone.openanolis.cn/ws/nrh4nnio/test_result/10491>:white_check_mark:SUCCESS</a></td>
    </tr>
    <tr>
        <td colspan="2">build-docker-image</td><td><a href=https://tone.openanolis.cn/ws/nrh4nnio/test_result/10491>:white_check_mark:SUCCESS</a></td>
    </tr>
    <tr>
        <td colspan="2">compile-nydus</td><td><a href=https://tone.openanolis.cn/ws/nrh4nnio/test_result/10491>:white_check_mark:SUCCESS</a></td>
    </tr>
    <tr>
        <td colspan="2">compile-ctr-remote</td><td><a href=https://tone.openanolis.cn/ws/nrh4nnio/test_result/10491>:white_check_mark:SUCCESS</a></td>
    </tr>
    <tr>
        <td colspan="2">compile-nydus-snapshotter</td><td><a href=https://tone.openanolis.cn/ws/nrh4nnio/test_result/10491>:white_check_mark:SUCCESS</a></td>
    </tr>
    <tr>
        <td colspan="2">start-nydus-snapshotter-config-containerd</td><td><a href=https://tone.openanolis.cn/ws/nrh4nnio/test_result/10491>:white_check_mark:SUCCESS</a></td>
    </tr>
    <tr>
        <td colspan="2">run-container-with-nydus-image</td><td><a href=https://tone.openanolis.cn/ws/nrh4nnio/test_result/10491>:white_check_mark:SUCCESS</a></td>
    </tr>
</table>

Congratulations, your test job passed!

File: src/bin/nydusd/fs_cache.rs (hunk 2 above)
Comment (benjaminhuanghuang) [reply src/bin/nydusd/fs_cache.rs:256]: @hsiangkao   You are right. I face this issue after cachefiles enabled. Do you know what are the reasons cause the cachefiles device issue?

**Comment by yqleng1987** [comment]
@benjaminhuanghuang , the code has been updated, so a new test job has been submitted. Please wait in patience. The test job url: [https://tone.openanolis.cn/ws/nrh4nnio/test_result/10858](https://tone.openanolis.cn/ws/nrh4nnio/test_result/10858)

**Comment by yqleng1987** [comment]
@benjaminhuanghuang , The CI test is completed, please check result:
<table>
    <tr>
        <th colspan="2">Test Case</th><th>Test Result</th>
    </tr>
    <tr>
        <td colspan="2">merge-target-branch</td><td><a href=https://tone.openanolis.cn/ws/nrh4nnio/test_result/10858>:white_check_mark:SUCCESS</a></td>
    </tr>
    <tr>
        <td colspan="2">build-docker-image</td><td><a href=https://tone.openanolis.cn/ws/nrh4nnio/test_result/10858>:white_check_mark:SUCCESS</a></td>
    </tr>
    <tr>
        <td colspan="2">compile-nydus</td><td><a href=https://tone.openanolis.cn/ws/nrh4nnio/test_result/10858>:white_check_mark:SUCCESS</a></td>
    </tr>
    <tr>
        <td colspan="2">compile-ctr-remote</td><td><a href=https://tone.openanolis.cn/ws/nrh4nnio/test_result/10858>:white_check_mark:SUCCESS</a></td>
    </tr>
    <tr>
        <td colspan="2">compile-nydus-snapshotter</td><td><a href=https://tone.openanolis.cn/ws/nrh4nnio/test_result/10858>:white_check_mark:SUCCESS</a></td>
    </tr>
    <tr>
        <td colspan="2">start-nydus-snapshotter-config-containerd</td><td><a href=https://tone.openanolis.cn/ws/nrh4nnio/test_result/10858>:white_check_mark:SUCCESS</a></td>
    </tr>
    <tr>
        <td colspan="2">run-container-with-nydus-image</td><td><a href=https://tone.openanolis.cn/ws/nrh4nnio/test_result/10858>:white_check_mark:SUCCESS</a></td>
    </tr>
</table>

Congratulations, your test job passed!

**Comment by yqleng1987** [comment]
@benjaminhuanghuang , the code has been updated, so a new test job has been submitted. Please wait in patience. The test job url: [https://tone.openanolis.cn/ws/nrh4nnio/test_result/10859](https://tone.openanolis.cn/ws/nrh4nnio/test_result/10859)

**Comment by yqleng1987** [comment]
@benjaminhuanghuang , the title has been updated, so a new test job has been submitted. Please wait in patience. The test job url: [https://tone.openanolis.cn/ws/nrh4nnio/test_result/10860](https://tone.openanolis.cn/ws/nrh4nnio/test_result/10860)

**Comment by yqleng1987** [comment]
@benjaminhuanghuang , the title has been updated, so a new test job has been submitted. Please wait in patience. The test job url: [https://tone.openanolis.cn/ws/nrh4nnio/test_result/10861](https://tone.openanolis.cn/ws/nrh4nnio/test_result/10861)

**Comment by yqleng1987** [comment]
@benjaminhuanghuang , The CI test is completed, please check result:
<table>
    <tr>
        <th colspan="2">Test Case</th><th>Test Result</th>
    </tr>
    <tr>
        <td colspan="2">merge-target-branch</td><td><a href=https://tone.openanolis.cn/ws/nrh4nnio/test_result/10859>:white_check_mark:SUCCESS</a></td>
    </tr>
    <tr>
        <td colspan="2">build-docker-image</td><td><a href=https://tone.openanolis.cn/ws/nrh4nnio/test_result/10859>:white_check_mark:SUCCESS</a></td>
    </tr>
    <tr>
        <td colspan="2">compile-nydus</td><td><a href=https://tone.openanolis.cn/ws/nrh4nnio/test_result/10859>:white_check_mark:SUCCESS</a></td>
    </tr>
    <tr>
        <td colspan="2">compile-ctr-remote</td><td><a href=https://tone.openanolis.cn/ws/nrh4nnio/test_result/10859>:white_check_mark:SUCCESS</a></td>
    </tr>
    <tr>
        <td colspan="2">compile-nydus-snapshotter</td><td><a href=https://tone.openanolis.cn/ws/nrh4nnio/test_result/10859>:white_check_mark:SUCCESS</a></td>
    </tr>
    <tr>
        <td colspan="2">start-nydus-snapshotter-config-containerd</td><td><a href=https://tone.openanolis.cn/ws/nrh4nnio/test_result/10859>:white_check_mark:SUCCESS</a></td>
    </tr>
    <tr>
        <td colspan="2">run-container-with-nydus-image</td><td><a href=https://tone.openanolis.cn/ws/nrh4nnio/test_result/10859>:white_check_mark:SUCCESS</a></td>
    </tr>
</table>

Congratulations, your test job passed!

**Comment by yqleng1987** [comment]
@benjaminhuanghuang , The CI test is completed, please check result:
<table>
    <tr>
        <th colspan="2">Test Case</th><th>Test Result</th>
    </tr>
    <tr>
        <td colspan="2">merge-target-branch</td><td><a href=https://tone.openanolis.cn/ws/nrh4nnio/test_result/10860>:white_check_mark:SUCCESS</a></td>
    </tr>
    <tr>
        <td colspan="2">build-docker-image</td><td><a href=https://tone.openanolis.cn/ws/nrh4nnio/test_result/10860>:white_check_mark:SUCCESS</a></td>
    </tr>
    <tr>
        <td colspan="2">compile-nydus</td><td><a href=https://tone.openanolis.cn/ws/nrh4nnio/test_result/10860>:white_check_mark:SUCCESS</a></td>
    </tr>
    <tr>
        <td colspan="2">compile-ctr-remote</td><td><a href=https://tone.openanolis.cn/ws/nrh4nnio/test_result/10860>:white_check_mark:SUCCESS</a></td>
    </tr>
    <tr>
        <td colspan="2">compile-nydus-snapshotter</td><td><a href=https://tone.openanolis.cn/ws/nrh4nnio/test_result/10860>:white_check_mark:SUCCESS</a></td>
    </tr>
    <tr>
        <td colspan="2">start-nydus-snapshotter-config-containerd</td><td><a href=https://tone.openanolis.cn/ws/nrh4nnio/test_result/10860>:white_check_mark:SUCCESS</a></td>
    </tr>
    <tr>
        <td colspan="2">run-container-with-nydus-image</td><td><a href=https://tone.openanolis.cn/ws/nrh4nnio/test_result/10860>:white_check_mark:SUCCESS</a></td>
    </tr>
</table>

Congratulations, your test job passed!

**Comment by yqleng1987** [comment]
@benjaminhuanghuang , The CI test is completed, please check result:
<table>
    <tr>
        <th colspan="2">Test Case</th><th>Test Result</th>
    </tr>
    <tr>
        <td colspan="2">merge-target-branch</td><td><a href=https://tone.openanolis.cn/ws/nrh4nnio/test_result/10861>:white_check_mark:SUCCESS</a></td>
    </tr>
    <tr>
        <td colspan="2">build-docker-image</td><td><a href=https://tone.openanolis.cn/ws/nrh4nnio/test_result/10861>:white_check_mark:SUCCESS</a></td>
    </tr>
    <tr>
        <td colspan="2">compile-nydus</td><td><a href=https://tone.openanolis.cn/ws/nrh4nnio/test_result/10861>:white_check_mark:SUCCESS</a></td>
    </tr>
    <tr>
        <td colspan="2">compile-ctr-remote</td><td><a href=https://tone.openanolis.cn/ws/nrh4nnio/test_result/10861>:white_check_mark:SUCCESS</a></td>
    </tr>
    <tr>
        <td colspan="2">compile-nydus-snapshotter</td><td><a href=https://tone.openanolis.cn/ws/nrh4nnio/test_result/10861>:white_check_mark:SUCCESS</a></td>
    </tr>
    <tr>
        <td colspan="2">start-nydus-snapshotter-config-containerd</td><td><a href=https://tone.openanolis.cn/ws/nrh4nnio/test_result/10861>:white_check_mark:SUCCESS</a></td>
    </tr>
    <tr>
        <td colspan="2">run-container-with-nydus-image</td><td><a href=https://tone.openanolis.cn/ws/nrh4nnio/test_result/10861>:white_check_mark:SUCCESS</a></td>
    </tr>
</table>

Congratulations, your test job passed!

**Pull request merged**